### ISSUE TITLE
Test tokio with other libraries

### ISSUE THREAD

**Issue by jettify** [PyO3/tokio#34]
-  https://github.com/aio-libs/aiorwlock/pull/35 passes
-  https://github.com/aio-libs/aiomonitor/pull/41 fails

**Comment by fafhrd91** [comment]
Awesome! Thanks!

Will check aiomonitor.

**Comment by jettify** [comment]
With `tokio` loop I am getting bunch of errors `RuntimeError: Timeout context manager should be used inside a task` when using `loop.run_until_complete(coro)` as result tests does not have a chance to start since such idiom ofter used in test fixtures. (asyncio and uvloop works as expected)

```
    @pytest.fixture
    def bucket_name(region, create_bucket, s3_client, loop):
>       name = loop.run_until_complete(create_bucket(region))
E       RuntimeError: Timeout context manager should be used inside a task
```

**Comment by fafhrd91** [comment]
This is because Task.current_task() does not work with tokio loop. Yury Selivanov will will info loop.current_task() to asyncio pep.
I added “loop.current_task()” to latest async_timeout and to aiohttp master.

**Comment by 1st1** [comment]
Does tokio create its own rust Tasks? Why no Futures?

**Comment by fafhrd91** [comment]
@jettify aiomonitor passes tests with aiohttp master

**Comment by fafhrd91** [comment]
- https://github.com/aio-libs/aioredis - tests pass. some xfail tests pass as well, which is not good

**Comment by jettify** [comment]
I will merge those PRs soon, so tokio/uvloop can be tested on each PR continuously.

**Comment by fafhrd91** [comment]
@jettify I think we need separate package for various asyncio related fixtures.
every library that test different event loops implements `loop` fixture

**Comment by jettify** [comment]
@fafhrd91 agreed there are few libraries, but not sure I like them (https://pypi.python.org/pypi/pytest-asyncio)

**Comment by fafhrd91** [comment]
But we have enough fixters in aiohttp and other aio libs, we just need to extract those to separate package
